auth/overview/api_key.go

```diff
@@ -0,0 +1,40 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package overview contains Google Cloud authentication overview snippets.
+// https://cloud.google.com/docs/authentication/
+package overview
+
+// [START auth_overview_api_key]
+import (
+	"context"
+	"fmt"
+
+	"google.golang.org/api/option"
+	"google.golang.org/api/pubsub/v1"
```

Comment:
Sorry I missed this in the second review - I don't think we should be using this import. This is an implicit recommendation of how to use PubSub. People should be using cloud.google.com/go/pubsub. This might match the other samples, but we should get these right from the start and update the others to be consistent after.

Can add options to https://godoc.org/cloud.google.com/go/pubsub#NewClient to use an API key, etc.

Reply:
Done.

Just curious, why we are providing two different libraries for the same API? How would the developers know which one to use (seems like a pretty confusing situation)?

Reply:
The google.golang.org one is auto-generated and the "raw" API and generally service-based clients. cloud.google.com/go is usually gRPC based. It's a mix of hand-written and auto-generated, but all around nicer to use in my opinion. If a cloud.google.com/go package exists, it's the one you should use.

If you have ideas about how to make it clear the cloud.google.com/go ones are preferred, we can make it happen. :)

Reply:
Okay, I see. Thanks for the explanation.